src/cairo.rs

```diff
@@ -75,8 +83,57 @@ pub fn run_cairo0(
         .unwrap();
 
     let program = Program::from_file(&prove_args.cairo_program, Some("main"))?;
+    let program_input = if let Some(program_input_file) = prove_args.program_input_file.clone() {
+        let program_input_file_str = std::fs::read_to_string(program_input_file)?;
+        let program_input =
+            serde_json::from_str::<HashMap<String, serde_json::Value>>(&program_input_file_str)?;
+        program_input
+    } else {
+        HashMap::new()
+    };
 
     let mut hint_processor = BuiltinHintProcessor::new_empty();
+    let program_input_clone = program_input.clone();
+    hint_processor.add_hint(
+        "ids.fibonacci_claim_index = program_input['fibonacci_claim_index']".to_string(),
+        Rc::new(HintFunc(Box::new(
+            move |vm, _exec_scopes, ids_data, ap_tracking, _constants| {
+                let fibonacci_claim_index = program_input
+                    .get("fibonacci_claim_index")
+                    .unwrap()
+                    .as_u64()
+                    .unwrap();
+                insert_value_from_var_name(
+                    "fibonacci_claim_index",
+                    Felt252::from(fibonacci_claim_index),
+                    vm,
+                    ids_data,
+                    ap_tracking,
+                )?;
+                Ok(())
+            },
+        ))),
+    );
+    hint_processor.add_hint(
+        "ids.iterations = program_input['iterations']".to_string(),
```

Comment:
I was wondering why the input has to be keyed "iterations", otherwise it throws error XD